pkg/ctl/cmdutils/cmd.go

```diff
@@ -40,6 +39,9 @@ func (c *Cmd) NewCtl() (*eks.ClusterProvider, error) {
 	}
 
 	for i, ng := range c.ClusterConfig.NodeGroups {
+		if err := api.SetKubeletExtraConfigDefaults(ng, c.ClusterConfig.Metadata); err != nil {
```

Comment:
This should be done later one when it is actually needed, not when we create the cluster provider. The cluster provider is created by many operations that don't need the extra kubelet configuration parameters so let's call this when it is actually needed (when we are compiling the user data). 

Something similar is done for bottlerocket, where we delayed the generation of the settings until the point where the user data has to be generated. Does this make sense? I think it will make testing a lot easier as well.